src/main/kotlin/blackjack/domain/deck/RandomDeckShuffleStrategy.kt

```diff
@@ -0,0 +1,10 @@
+package blackjack.domain.deck
+
+import blackjack.domain.card.Card
+import kotlin.random.Random
+
+class RandomDeckShuffleStrategy : DeckShuffleStarategy {
+    override fun shuffleDeck(deck: MutableList<Card>) {
+        deck.shuffle(Random(System.currentTimeMillis()))
```

Comment:
```suggestion
        return deck.shuffled(Random(System.currentTimeMillis()))
```
파라미터의 타입이 MutableList로 되어있고, shuffle 함수를 통해서 원본을 변경하고 있네요.
원본을 변경하게 되면, 원본을 참조하는 곳에서 예상치 못한 동작을 하게 됩니다.
방어적 복사를 활용해서 원본을 손상시키지 않도록 개선해보면 어떨까요?

Reply:
이 부분은 잘 이해가 되지 않아서 질문 남겨봅니다~

제가 알기론 방어적 복사는 객체 내부의 특정 변수를 외부에서 마음대로 건드리지 못하도록 막으려는 의도로 하는것으로 아는데요
shuffleDeck 메서드를 Player의 shuffleDeck() 메서드에서만 호출하고 있고 Player 객체 생성시에 딱 한번만 호출하고 있습니다.
이 경우에도 방어적 복사를 적용하는게 맞을까요?

Reply:
종현님이 말씀하신 방법도 좋은 방법이며, 현 구조에서는 문제가 없어보입니다. 👍 
다만, 가급적이면 원본을 바꾸지 않는 방법으로 코드를 작성하시는 것을 지향하셨으면 좋겠어서 코맨트를 작성했습니다. 🙏 

원본을 바꾸게 되는 경우, 어떤 일이 발생하게 되는지 예시를 들어드리자면 아래와 같습니다.
```kotlin
// 이해를 돕기 위한 코드이니, 디테일이 부족한 부분은 양해부탁드려요!

class Test(val deck: List<Card>)

val deck = mutableListOf<Card>()
deck.add(Card(1))
deck.add(Card(2))
val test = Test(deck = deck)

RandomDeckShuffleStrategy().shuffleDeck(deck)
```
위 경우, `test.deck.first()`는 반드시 Card(1)일 것이라 장담할 수 있을까요? 🤔